### PR TITLE
feat: add env endpoint for server

### PR DESCRIPTION
**Pull request by Spazzy757**
Changelog

- adds a `env`  to the `/info` endpoint for the server

Fixes: https://github.com/sharat87/httpbun/issues/7

### CODE REVIEW

File: bun/bun.go [hunk 1]
```diff
@@ -21,6 +17,11 @@ import (
 	"strings"
 	"time"
 	"unicode/utf8"
+
+	"github.com/sharat87/httpbun/assets"
+	"github.com/sharat87/httpbun/exchange"
+	"github.com/sharat87/httpbun/mux"
+	"github.com/sharat87/httpbun/util"
```
Comment (sharat87) [inline bun/bun.go:24]: Can we move these imports back to where they were please? Thanks!

File: bun/bun.go [hunk 2]
```diff
@@ -74,6 +75,7 @@ func MakeBunHandler(pathPrefix, commit, date string) mux.Mux {
 	m.HandleFunc("/html", handleSampleHtml)
 	m.HandleFunc("/json", handleSampleJson)
 	m.HandleFunc("/robots.txt", handleSampleRobotsTxt)
+	m.HandleFunc("/env", handleEnv)
```
Comment (sharat87) [inline bun/bun.go:78]: As discussed on the linked issue, let's move this into `/info` and remove this line.

File: bun/bun.go [hunk 3]
```diff
@@ -497,3 +499,14 @@ func handleInfo(ex *exchange.Exchange) {
 		"hostname": hostname,
 	})
 }
+
+func handleEnv(ex *exchange.Exchange) {
+	ex.ResponseWriter.Header().Set("Content-Type", "application/json")
+	data := make(map[string]interface{})
+	for _, e := range os.Environ() {
+		pair := strings.SplitN(e, "=", 2)
+		data[pair[0]] = os.Getenv(pair[0])
```
Comment (sharat87) [inline bun/bun.go:508]: The [`strings.Cut`](https://pkg.go.dev/strings#Cut) would read better I think, no?

```suggestion
		name, value, _ := strings.Cut(e, "=")
		data[name] = value
```

File: bun/bun.go [hunk 4]
```diff
@@ -497,3 +499,14 @@ func handleInfo(ex *exchange.Exchange) {
...
+	}
+	jsonContent := util.ToJsonMust(data)
+	ex.WriteBytes(jsonContent)
```
Comment (sharat87) [inline bun/bun.go:511]: The `WriteJson` method might be simpler, and we don't have to set the `Content-Type` header as well.

```suggestion
	util.WriteJson(ex.ResponseWriter, data)
```

File: bun/bun.go [hunk 5]
```diff
@@ -497,3 +499,14 @@ func handleInfo(ex *exchange.Exchange) {
 		"hostname": hostname,
 	})
 }
+
+func handleEnv(ex *exchange.Exchange) {
+	ex.ResponseWriter.Header().Set("Content-Type", "application/json")
+	data := make(map[string]interface{})
```
Comment (sharat87) [inline bun/bun.go:505]:
```suggestion
	data := make(map[string]any)
```

**Review by sharat87 — COMMENTED** [review]
Few minor things in addition to merging this with `/info`, and we should be good!

File: bun/bun.go (hunk 1 above)
Comment (Spazzy757) [reply bun/bun.go:24]: @sharat87 This is from [goimports](https://cs.opensource.google/go/x/tools/+/master:README.md), I can move them back, however this from my understanding of go and the defacto tools library is how imports should be formatted. Here is [the commit](https://go-review.googlesource.com/c/tools/+/25145) that introduces this to the Golang org

Comment (sharat87) [reply bun/bun.go:24]: Hey, thanks for coming back on this quick.

1. I realize this is from `goimports`, but the current formatting is from just running `go fmt`. This is a debatable point on which one is better, but I don't see much value in changing what we already have. I'd prefer to stick with status quo here.
2. If we indeed want to change import ordering to using `goimports`, I don't think this is the PR for this. Ideally, we should do it in _all_ source files at once, so we're consistent across the codebase, and we should do it as part of a dedicated PR just for solving that problem. Not that this is something I'm inclined to doing, but just looking to keep the PR focused on just the problem being solved.

Does that sound okay?

Comment (Spazzy757) [reply bun/bun.go]: Your repos, your rules :+1:

**Comment by sharat87** [comment]
Thank you @Spazzy757, very appreciate you engaging in the discussion to get this merged. Thank you so much for your contribution! 🙏

**Pull request merged**